driver-core/src/main/com/mongodb/event/ConnectionCheckOutFailedEvent.java

```diff
@@ -75,6 +98,27 @@ public ServerId getServerId() {
         return serverId;
     }
 
+    /**
+     * Gets the operation identifier
+     *
+     * @return the operation identifier
+     * @since 4.10
+     */
+    public long getOperationId() {
+        return operationId;
+    }
+
+    /**
+     * Gets the elapsed time between check out started and check out failed.
+     *
+     * @param timeUnit the time unit
+     * @return the elapsed time
+     * @since 4.10
+     */
+    public long getElapsedTime(final TimeUnit timeUnit) {
```

Comment:
Producing `Duration` is better than producing a unitless value. I imagine, the current approach is chosen to remain consistent with the rest of the API. I propose to change all the duration-accepting/producing API in 5.0. If that seems like a good idea, I'll file a ticket.

Reply:
Removed elapsedTime from PR